source/vmix_kpp.F90

```diff
@@ -404,22 +398,21 @@ subroutine init_vmix_kpp(CVmix_vars, VDC, VVC, convect_diff, convect_visc)
       write(stdout,fmt_log ) '  use CVMix KPP routines    =', lcvmix
 
       select case (langmuir_opt)
-      ! QL, 150612, set llangmuir_efactor and lenhanced_entrainment based on langmuir_opt 
       case ('null')
          write(stdout,'(a38)') '   no Lanmguir mixing parameterization'
-         llangmuir_efactor      = .false.
-         lenhanced_entrainment  = .false.
+         langmuir_mixing_opt      = 'NONE'
+         langmuir_entrainment_opt = 'NONE'
       case ('vr12-ma')
          write(stdout,'(a38)') '   Langmuir param. option: vr12-ma    '
-         llangmuir_efactor      = .true.
-         lenhanced_entrainment  = .false.
-      case ('vr12-en')
-         write(stdout,'(a38)') '   Langmuir param. option: vr12-en    '
-         llangmuir_efactor      = .true.
-         lenhanced_entrainment  = .true.
+         langmuir_mixing_opt      = 'LWF16'
+         langmuir_entrainment_opt = 'LWF16'
+      case ('lf17')
+         write(stdout,'(a38)') '   Langmuir param. option: lf17       '
+         langmuir_mixing_opt      = 'LWF16'
+         langmuir_entrainment_opt = 'LF17'
       case default
-         llangmuir_efactor      = .false.
-         lenhanced_entrainment  = .false.
+         langmuir_mixing_opt      = 'NONE'
```

Comment:
Why is the default behavior here to silently fallback to `langmuir_opt = 'null'`? I would expect an abort with a message that the provided `langmuir_opt` is not supported

Reply:
Now it should abort if an invalid option is used.